File: LeetCode/Simulation/JS/Leetcode537-复数乘法.js
var complexNumberMultiply = function(num1, num2) {
    const c1 = [num1.split('+')[0],num1.split('+')[1].split('i')[0]];
    const c2 = [num2.split('+')[0],num2.split('+')[1].split('i')[0]];
    const r1 = parseInt(c1[0]),i1 = parseInt(c1[1]);
    const r2 = parseInt(c2[0]),i2 = parseInt(c2[1]);
    return '' + r1 * r2 - i1 * i2 + '+' + (r1 * i2 + r2 * i1) + 'i';
}

File: LeetCode/Simulation/Golang/Leetcode537-复数乘法.go
package sbgo

import (
	"fmt"
	"strconv"
	"strings"
)

func complexNumberMultiply(num1 string, num2 string) string {
	r1, i1 := parse(num1)
	r2, i2 := parse(num2)
	return fmt.Sprintf("%d+%di", r1*r2-i1*i2, r1*i2+r2*i1)
}

func parse(s string) (int, int) {
	idx := strings.IndexByte(s, '+')
	r, _ := strconv.Atoi(s[:idx])
	i, _ := strconv.Atoi(s[idx+1 : len(s)-1])
	return r, i
}
